src/components/AddRecipe.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react';
+import RecipeService from '../services/RecipeService';
+
+const AddRecipe = () => {
+  const initialRecipeState = {
+    id: null,
+    name: '',
+    description: '',
+    ingredients: '',
+  };
+
+  const [recipe, setRecipe] = useState(initialRecipeState);
+  const [submitted, setSubmitted] = useState(false);
+
+  const handleInputChange = event => {
+    const { name, value } = event.target;
+    setRecipe({ ...recipe, [name]: value });
+  };
+
+  const saveRecipe = () => {
+    const ingredients = recipe.ingredients
+      .split(',')
+      .map(ingredient => ({ name: ingredient.trim() }));
+
+    const data = {
+      name: recipe.name,
+      description: recipe.description,
+      ingredients: ingredients,
+    };
+
+    RecipeService.create(data)
```

Comment:
Whenever possible, let's try to use [async/await syntax ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function)(the default in travelperk) for dealing with async calls. This makes for a simpler syntax in most cases